### DISCUSSION TITLE
Authentication With LDAP

### DISCUSSION THREAD

**Discussion by netlabit** [seriousm4x/UpSnap#76] [Q&A]
it would be very interesting if it could authenticate with LDAP, do you think you can integrate it?

**Comment by seriousm4x** [comment]
Here is an upstream feature request for LDAP: https://github.com/pocketbase/pocketbase/issues/971

It looks like it might be possible:

> External auth integrations different than OAuth2, at least for v1.0.0, are not planned but users can add support for that on their own by registering a custom middleware

But I have never used LDAP before. I'd prefer waiting for an upstream implementation and then integrating it here.

**Reply by netlabit** [reply]
Hu
i dropped ldap in favor of oauth2 with authentik

when i call http://mydomain.it:8090/api/oauth2-redirect, 

i have 

{"code":404,"message":"Not Found.","data":{}}

**Comment by toras9000** [comment]
I am not very knowledgeable in this area, but I will share the methods I have used.

This should be possible by using an IdP that can be linked to an LDAP server.
Using a combination of `OpenLDAP` and `Keycloak`, we were able to authenticate using LDAP users as follows

- Setup LDAP federation with `Keycloak` user federation
- Define OIDC settings for UpSnap in `Keycloak` client settings
- Configure OIDC authentication in Auth providers setting in UpSpan (PocketBase)
